txscript/opcode.go

```diff
@@ -2575,7 +2575,16 @@ func opcodeCheckMultiSig(op *opcode, data []byte, vm *Engine) error {
 		return scriptError(ErrTooManyOperations, str)
 	}
 
-	pubKeys := make([][]byte, 0, numPubKeys)
+	// Use an array on the stack for typical sizes to avoid an allocation in the
+	// vast majority of cases.
+	const numPubKeysHint = 5
+	var pubKeys [][]byte
+	if numPubKeys <= numPubKeysHint {
+		var pubKeysArr [numPubKeysHint][]byte
+		pubKeys = pubKeysArr[0:0:numPubKeysHint]
```

Comment:
This could just be `numPubKeys`, right?

```suggestion
		pubKeys = pubKeysArr[0:0:numPubKeys]
```

Reply:
Good call.  I originally had it outside of the if where it had to be the hint since `numPubKeys` could've been larger, but since it's bounds checked now, it can be further limited down to the actual number of keys, the same as done below with the signatures.

Updated.